test-adapter/src/main/scala/org/scalajs/sbttestadapter/TestAdapter.scala

```diff
@@ -58,49 +61,60 @@ final class TestAdapter(jsEnv: ComJSEnv, config: TestAdapter.Config) {
   }
 
   /** Releases all resources. All associated runs must be done. */
-  def close(): Unit = {
-    if (!closing.getAndSet(true)) {
-      // Snapshot runs.
-      val seenRuns = runs.keySet
+  def close(): Unit = synchronized {
+    stopEverything()
+    if (runs.nonEmpty) {
+      config.logger.warn(s"TestAdapter.close() called with incomplete runs: $runs")
+    }
+  }
+
+  /** Called when a throwable bubbles up the execution stack.
+   *
+   *  We terminate everyting if this happens to make sure nothing hangs waiting
+   *  on an async operation to complete.
+   */
+  private def reportFailure(cause: Throwable): Unit = {
+    config.logger.error("Failure in async execution. Aborting all test runs.")
+    config.logger.trace(cause)
+    stopEverything()
```

Comment:
IIUC, this causes the `close()` methods of the Com's to be caused, which, due to the added `ComJSRPC.super.close()`, ends up calling `failing.promise.failure(new IOException("Channel got closed"))` in `RPCCore.close()`, which is what trickles into actually failing the sbt test interface waiting for that future. Is that right?

Reply:
Yep. Exactly.